vscode/vscode/vscode.ml

```diff
@@ -2521,6 +2652,271 @@ module TreeView = struct
+end
+
+module WebviewOptions = struct
+  include Interface.Make ()
+
+  include
+    [%js:
+    val enableCommandUris : t -> bool [@@js.get]
+
+    val enableScripts : t -> bool [@@js.get]
+
+    val set_enableScripts : t -> bool -> unit [@@js.set]
+
+    val localResourceRoots : t -> Uri.t list [@@js.get]
+
+    val portMapping : t -> WebviewPortMapping.t list [@@js.get]
+
+    val create :
+         enableCommandUris:bool
+      -> enableScripts:bool
+      -> localResourceRoots:Uri.t list
+      -> portMapping:WebviewPortMapping.t list
+      -> t
+      [@@js.builder]]
+end
+
+module WebView = struct
+  include Interface.Make ()
+
+  module OnDidReceiveMessage = Event.Make (Js.Any)
+
+  include
+    [%js:
+    val onDidReceiveMessage : t -> OnDidReceiveMessage.t [@@js.get]
+
+    val cspSource : t -> string [@@js.get]
+
+    val html : t -> string [@@js.get]
+
+    val set_html : t -> string -> unit [@@js.set]
+
+    val options : t -> WebviewOptions.t [@@js.get]
+
+    val set_options : t -> WebviewOptions.t -> unit [@@js.set]
+
+    val asWebviewUri : t -> localResource:Uri.t -> Uri.t [@@js.call]
+
+    val postMessage : t -> Js.Any.t -> bool Promise.t [@@js.call]
+
+    val create :
+         onDidReceiveMessage:OnDidReceiveMessage.t
+      -> cspSource:string
+      -> html:string
+      -> options:WebviewOptions.t
+      -> close:(unit -> unit)
+      -> asWebviewUri:(Uri.t -> Uri.t)
+      -> postMessage:(Js.Any.t -> bool Promise.t)
+      -> t
+      [@@js.builder]]
+end
+
+module rec WebviewPanel : sig
+  include Js.T
+
+  module LightDarkIcon : sig
+    type t =
+      { light : [ `Uri of Uri.t ]
+      ; dark : [ `Uri of Uri.t ]
+      }
+
+    include Js.T with type t := t
+  end
+
+  val onDidChangeViewState :
+    t -> WebviewPanelOnDidChangeViewStateEvent.t Event.t
+
+  val onDidDispose : t -> unit Event.t
+
+  val active : t -> bool
+
+  type iconPath =
+    [ `Uri of Uri.t
+    | `LightDark of LightDarkIcon.t
+    ]
+
+  val options : t -> WebviewPanelOptions.t
+
+  val title : t -> string
+
+  val viewColumn : t -> ViewColumn.t
+
+  val viewType : t -> string
+
+  val visible : t -> bool
+
+  val webview : t -> WebView.t
+
+  val set_webview : t -> WebView.t -> unit
+
+  val dispose : t -> Js.Any.t
+
+  val reveal :
+    t -> ?preserveFocus:bool -> ?viewColumn:ViewColumn.t -> unit -> unit
+
+  val create :
+       onDidChangeViewState:WebviewPanelOnDidChangeViewStateEvent.t Event.t
+    -> onDidDispose:unit Event.t
+    -> active:bool
+    -> options:WebviewPanelOptions.t
+    -> title:string
+    -> viewColumn:ViewColumn.t
+    -> viewType:string
+    -> visible:bool
+    -> webview:WebView.t
+    -> dispose:Js.Any.t
+    -> reveal:(?preserveFocus:bool -> ?viewColumn:ViewColumn.t -> unit -> unit)
+    -> t
+end = struct
+  include Interface.Make ()
+
+  module LightDarkIcon = struct
+    type t =
+      { light : ([ `Uri of Uri.t ][@js.union])
+      ; dark : ([ `Uri of Uri.t ][@js.union])
```

Comment:
Making these polymorphic variants seems unnecessary. Can't it just be `: Uri.t`? It would remove the need for the custom `t_of_js` function.

Reply:
It is indeed unnecessary, fixed in 7efe612f6bc6db5f6773d5b4fa8fdc0c089bce49.